### PR TITLE
Unit tests for Vec3 + fixes

### PR DESCRIPTION
**Pull request by einarf**
* Fix up Vec3 to match Vec2
* Create extensive unit tests Vec3
* Fix up Vec4 to match Vec2
* Create extensive unit tests for Vec4
* Fix various small things here and there

### CODE REVIEW

**Review by benmoran56 — APPROVED** [review]

**Review by benmoran56 — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -40,10 +41,12 @@ ignore = [
     "TRY",  # no triceratops.
     "NPY",  # remove numpy
     "Q000",  # double quotes
+    "EM101",  # String literals in exception
+    "B028",  # No explicit stack level in warn
+    "FIX002",  # Fixme
+    "TD003",  # Missing issue link on the line following this TODO"
 ]
 
-exclude = ["venv/*", ".venv/*", "build/*", "doc/*", ".github/*"]
```
Comment (caffeinepills) [inline pyproject.toml:45]: This should be kept, any reason to not exclude these?

File: pyproject.toml [hunk 2]
```diff
@@ -58,9 +61,6 @@ inline-quotes = "single"
 quote-style = "preserve"
 indent-style = "space"
 
-[tool.ruff.flake8-quotes]
-inline-quotes = 'single'
```
Comment (caffeinepills) [inline pyproject.toml:62]: I think Benjamin decided on single quotes for the project, unless I'm mistaken.

**Review by benmoran56 — APPROVED** [review]

File: pyproject.toml (hunk 2 above)
Comment (benmoran56) [reply pyproject.toml]: This PR was behind the current development branch, but I manually resolved the conflicts on my end.

Comment (einarf) [reply pyproject.toml]: This was just removing a duplicate section.

File: pyproject.toml (hunk 1 above)
Comment (einarf) [reply pyproject.toml]: I was just moving it up to the main section

Comment (einarf) [reply pyproject.toml]: Doubled checked the merge. `exclude` was indeed moved to the ruff main secion.
```toml
[tool.ruff]
exclude = ["venv/*", ".venv/*", "build/*", "doc/*", ".github/*", "tests"]
```

File: pyproject.toml (hunk 2 above)
Comment (einarf) [reply pyproject.toml]: Double checked the merge. The removed section is deprecated and we already have this
```toml
[tool.ruff.lint.flake8-quotes]
inline-quotes = "single"
```

**Pull request merged**